weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx

```diff
@@ -64,6 +64,40 @@ const getRefs = (data: Data): string[] => {
 
 type RefValues = Record<string, any>; // ref URI to value
 
+// Returns a ref or undefined from a path like "value.something[0].nested" or "value.nested" and an object
+const getRefFromObjectPath = (obj: any, path: string) => {
```

Comment:
When i see path parsing like this, I hear warning bells. It seems like this is fragile and leveraging a contract that is not explicit. I think we can do better here. Specifically, the expandId is the stringified id from the traverse module. i'd recommend creating a single useEffect:

```typescript
  // This effect will ensure that all "expandedIds" whose value is a ref
  // have the ref added to the `expandedRefs` state.
  useEffect(() => {
    const expandRefsToAdd: {[ref: string]: string} = {};
    const mapper = (context: TraverseContext) => {
      const contextPath = context.path.toString();
      if (
        expandedIds.includes(contextPath) &&
        isWeaveRef(context.value) &&
        expandedRefs[context.value] == null
      ) {
        expandRefsToAdd[context.value] = contextPath;
      }
      return context.value
    };
    mapObject(data, mapper);
    setExpandedRefs(eRefs => ({...eRefs, ...expandRefsToAdd}));
  }, [data, expandedIds, expandedRefs, setExpandedRefs]);
```

I think something like this will solve the problem